spirv_glsl.cpp

```diff
@@ -14865,11 +14865,11 @@ void CompilerGLSL::emit_block_chain(SPIRBlock &block)
 
 		size_t num_blocks = block_declaration_order.size();
 
-		const auto to_case_label = [](uint32_t literal, bool is_unsigned_case) -> string {
-			return is_unsigned_case ? convert_to_string(literal) : convert_to_string(int32_t(literal));
+		const auto to_case_label = [](uint64_t literal, bool is_unsigned_case) -> string {
+			return is_unsigned_case ? convert_to_string(literal) : convert_to_string(int64_t(literal));
```

Comment:
This seems broken for 32-bit int case literals, we won't get sign extension I think?

Reply:
I'll write some tests to check that!

Reply:
There's also the cases below for line 14913 where we have to emit some branches when "complex OpPhi" is used. There are some generated conditionals there that might be affected with 64-bit switch.

Reply:
It uses to_case_label, so it *should* be fine, but good to sanity check it ...

Reply:
Just wrote a quick test and you're right, we're loosing sign extension for 32bit values. I'll include the test in the next commit too.